### PR TITLE
Making homepage search components load data dynamically

### PR DESCRIPTION
**Pull request by mantagen**
- activities and categories load from server
- categories asks user to select an activity if none selected

### CODE REVIEW

File: client/src/ActivitySelect.elm [hunk 1]
```diff
@@ -191,7 +164,12 @@ view model inputClass =
                 ]
 
         buttonClass =
-            inputClass ++ " tl truncate bg-white " ++ classNames [ ( "bg-blue white", menuOpen ), ( "black-60", selectedActivity.name == emptyActivity.name ) ]
+            "w-100 h2 fl pv2 ph3 br-pill ba b--solid b--blue"
+                ++ " tl truncate bg-white "
```
Comment (Jbarget) [inline client/src/ActivitySelect.elm]: do you need the `++` here? can just add it to the previous string

File: client/src/CategorySelect.elm [hunk 2]
```diff
@@ -132,73 +119,86 @@ view model =
...
+                    , onBlur HandleMenuBlur
+                    , tabindex -1
+                    ]
+                    [ div [ class "ph4 pv2 f6" ] [ text "Please select an Activity" ] ]
+                )
+            , viewIf (atLeastOneOption)
+                (div
+                    [ id "category-list"
+                    , class menuClass
+                    , ariaExpanded (String.toLower (toString menuOpen))
```
Comment (Jbarget) [inline client/src/CategorySelect.elm]: i think we have a util for this

File: client/src/CategorySelect.elm [hunk 3]
```diff
@@ -132,73 +119,86 @@ view model =
...
-                                    ]
-
-                            optionBgClass =
-                                classNames
-                                    [ ( "absolute absolute--fill", True )
-                                    , ( "bg-blue o-20", isFocused )
+            , viewIf (not atLeastOneOption)
+                (div
+                    [ class menuClass
+                    , ariaExpanded (String.toLower (toString menuOpen))
```
Comment (Jbarget) [inline client/src/CategorySelect.elm]: util for this

File: client/src/Update/Update.elm [hunk 4]
```diff
@@ -61,5 +79,35 @@ update msg model =
...
+                    , countries = results.countries
+                    , activitySelect = newActivityModel
+                    , categorySelect = newCategoryModel
+                    , countrySelect = newCountryModel
+                  }
+                , Cmd.none
+                )
+
+        ComponentData (Err _) ->
+            ( model, Cmd.none )
```
Comment (Jbarget) [inline client/src/Update/Update.elm]: we should ask Noga for an example error state

**Comment by Jbarget** [comment]
screen shot 2018-06-17 at 20 18 05

Set a max height on the dropdown and make it alphabetical A-Z

**Comment by Jbarget** [comment]
Also, not all the countries have flags for me, should they? Also spotted a weird country "East Africa". there may be more...

**Comment by mantagen** [comment]
@Jbarget re flags, and non-countries, I spoke with @KieranGarvey about this and he said leave it for now, but we'll have to adjust the countries list.

**Pull request merged**